contract/src/main/kotlin/io/provenance/scope/loan/utility/LoanContractValidations.kt

```diff
@@ -0,0 +1,64 @@
+package io.provenance.scope.loan.utility
+
+import io.dartinc.registry.v1beta1.ENote
+import tech.figure.loan.v1beta1.LoanDocuments
+import tech.figure.servicing.v1beta1.ServicingRightsOuterClass.ServicingRights
+import tech.figure.util.v1beta1.DocumentMetadata
+import io.dartinc.registry.v1beta1.Controller as ENoteController
+
+internal val documentListInputValidation: (LoanDocuments) -> Unit = { documents ->
+    validateRequirements(ContractRequirementType.VALID_INPUT) {
+        documents.documentList.forEach { document ->
+            documentValidation(document)
+        }
+    }
+}
+
+internal val documentValidation: ContractEnforcementContext.(DocumentMetadata) -> Unit = { document ->
```

Comment:
I'm pretty new to Kotlin, but could we add extension functions to each of these types, so DocumentMetadata.validate() could be called?

Reply:
So IMO we'd only want to use extension functions for validating data if our usecase was simpler. However, I would point to the signatures of the values in this file as the reason why I think my approach is better - using lambda functions allows us to compound validation rules (e.g. an eNote validation is composed of an eNote document validation and an eNote controller validation, each of which are also used on their own in other contracts) and take advantage of the `ContractEnforcementContext` to still consolidate input violations into a single message.